### PR TITLE
Feature/simplify interaction with deployed artefacts

### PR DESCRIPTION
**Pull request by matthiaszimmermann**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,21 +255,85 @@ brownie run scripts/instance.py dump_sources 0x2852593b21796b549555d09873155B252
...
+`instanceOperator` has sufficient funds to cover the setup.
+
+For testnets faucet funds may be used
+
+* [avax-test (Fuji (C-Chain))](https://faucet.avax.network/)
+* [polygon-test](https://faucet.polygon.technology/)
+
+Using the ganache scenario shown below ensures that all addresses used are sufficiently funded.
+
 ```bash
```
Comment (doerfli) [inline README.md:268]: Why bash and not python?

File: scripts/deploy.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
+from brownie.network import accounts
```
Comment (doerfli) [inline scripts/deploy.py:1]: Since this script is used to deploy the ayii product we should rename is `deploy_ayii.py`

File: scripts/deploy.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    ayiiProduct = ayiiDeploy.getProduct()
+    ayiiOracle = ayiiProduct.getOracle()
+    ayiiRiskpool = ayiiProduct.getRiskpool()
+
+    product = ayiiProduct.getContract()
+    oracle = ayiiOracle.getContract()
+    riskpool = ayiiRiskpool.getContract()
+
+    # investor funding and bundle creation
+    bundleInitialFunding=1000000
```
Comment (doerfli) [inline scripts/deploy.py:164]: move to constant

File: scripts/deploy.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            print('returning last product available')
+        
+        componentId = instanceService.getProductId(products-1)
+        component = instanceService.getComponent(componentId)
+        product = contract_from_address(AyiiProduct, component)
+    else:
+        print('1 product expected, no producta available')
+        print('no product returned (None)')
+
+    if oracles >= 1:
```
Comment (doerfli) [inline scripts/deploy.py:272]:
```python

    if oracles > 1:
        print('1 oracle expected, {} oracles available'.format(oracles))
        print('returning last oracle available')
    if oracles >= 1:
```

File: scripts/deploy.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+    products = instanceService.products()
+    oracles = instanceService.oracles()
+    riskpools = instanceService.riskpools()
+
+    product = None
+    oracle = None
+    riskpool = None
+
+    if products >= 1:
```
Comment (doerfli) [inline scripts/deploy.py:260]:
```python
    if products > 1:
        print('1 product expected, {} product available'.format(products))
        print('returning last product available')
    if products >= 1:
```

File: scripts/deploy.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+            print('returning last oracle available')
+        
+        componentId = instanceService.getOracleId(oracles-1)
+        component = instanceService.getComponent(componentId)        
+        oracle = contract_from_address(AyiiOracle, component)
+    else:
+        print('1 oracle expected, no oracles available')
+        print('no oracle returned (None)')
+
+    if riskpools >= 1:
```
Comment (doerfli) [inline scripts/deploy.py:284]:
```python
    if riskpools > 1:
        print('1 riskpool expected, {} riskpools available'.format(riskpools))
        print('returning last riskpool available')
    if riskpools >= 1:
```

File: scripts/deploy.py [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+            applicationFilter, 
+            bundleInitialFunding, 
+            {'from': investor})
+
+    # create risks
+    projectId = s2b32('2022.kenya.wfp.ayii')
+    uaiId = [s2b32('1234'), s2b32('2345')]
+    cropId = s2b32('mixed')
+    
+    triggerFloat = 0.75
```
Comment (doerfli) [inline scripts/deploy.py:183]: move those 4 to constants as well

**Review by doerfli — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (matthiaszimmermann) [reply README.md:268]: done

File: scripts/deploy.py (hunk 2 above)
Comment (matthiaszimmermann) [reply scripts/deploy.py:1]: done

File: scripts/deploy.py (hunk 3 above)
Comment (matthiaszimmermann) [reply scripts/deploy.py:164]: no, prefer to see the actual value where it is used in this case

**Pull request merged**